### PR TITLE
Remove styling for .form-control:focus:invalid

### PR DESCRIPTION
**Pull request by npvn**
With these rules, when users click on an input with specific type (e.g. url, email,...), even if they haven't start typing, the input will be turned red. This is because the input is invalid (in this case, empty) at the focus moment. Demonstration: http://jsfiddle.net/9tndpu8o

This is not a good UX practice. Users properly want their inputs validated only after an event like blur or submit.

I suggest that we remove these two rules. If a developer need to highlight an invalidated input, s/he can add `has-error` class to the input's `.form-group` container on a more appropriate event like blur or submit.

### CODE REVIEW

File: dist/css/ripples.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
+.withripple {
```
Comment (FezVrasta) [inline dist/css/ripples.css:1]: this file should not exists according to the new grunt task

**Comment by FezVrasta** [comment]
This is exactly the same behavior proposed by Polymer so I think we'll keep it for now.

http://www.polymer-project.org/components/paper-elements/demo.html#paper-input

**Comment by npvn** [comment]
Hmm, I think it's not really accurate to say that this behavior is intended by the Polymer team.

The paper elements section on Polymer website is just a demonstration of Material Design guideline. Most of it follow the doc strictly, but we shoudn't assume that it's a perfect doc alternative. In fact, even the Material Design website still has flaws: http://blog.creative-tim.com/how-google-failed-at-implementing-its-own-material-design/

The best place to find out what are proposed by the Material Design team is the doc. If the doc mentions something, then we're sure it's intended, and we'll follow it strictly.

If the doc doesn't mention it, then we need to use our best judgement. To come up with the best judgement we can take a look at Polymer (and other Material Design implementation/demonstration), but **as a reference only**. Eventually the solution should be what we decided as the best.

The best judgement in this case should be obvious from a UX point of view.

**Comment by FezVrasta** [comment]
Check this pen:
http://codepen.io/FezVrasta/pen/qfGJI

If you focus the input and don't write anything, the color is the default one, once you start writing it becomes red until you write a valid text. So why do you say that it turns red even when empty?

**Comment by npvn** [comment]
The behavior I described will happen when we add `required` attribute to the input element: http://codepen.io/npvn/pen/dBypj

Is there any disadvantage if we remove the two rules I mentioned?

**Comment by FezVrasta** [comment]
Actually it's good that it's red if is empty and required... I can't really see where is the bad ux

**Comment by npvn** [comment]
Hmm, looks like there's no 'best' solution for this since it depends quite strongly on personal perception.

I think validation should happen only when user has finish entering input (on blur or submit). When I click on an input field and see it turns red, I wonder what's wrong, since red is an indicator of "I'm doing something wrong" and I'm surprised because I haven't done anything yet. But this may just be my personal opinion.

When the best solution is unclear like this, do you think we should allow each developer to decide his own solution? 

If we remove the two mentioned rules, developers wanting the input field to be red on focus can add necessary `.form-control:focus:invalid` styles.

On the other hand, if we keep those rules, developers thinking like me would have no choice.

Sorry for drawing a lot of your time and energy to discuss a subtle problem like this :D.

**Comment by npvn** [comment]
@FezVrasta How is it going?

If this is not what you want to discuss at the moment, it's ok, just let me know so I can start thinking about other solutions.

**Comment by FezVrasta** [comment]
This is the default behaviour of the HTML specifications, I think we should keep in this way... If more people will ask this change I'll do something

**Pull request closed without merging**